Updated docs about using ROSE_DATAC (#736)
cylc/cylc-doc: src/workflow-design-guide/general-principles.rst

paths should be under the *workflow share directory*, the location of which is
passed to all tasks as ``$CYLC_WORKFLOW_SHARE_DIR``.

The ``$CYLC_SHARE_CYCLE_DIR`` variable can be used to target
cyclepoint-specific locations under the workflow share directory.

Sometimes it may be useful to refer to other cycles - to do this use
``isodatetime``:

.. code-block:: bash

   # Cylc task script
   CYCLE_POINT_MINUS_P1D=$(isodatetime "${CYLC_TASK_CYCLE_POINT} --offset -P1D)"
   SHARE_CYCLE_DIR_MINUS_P1D="${CYLC_WORKFLOW_SHARE_DIR}/cycle/${CYCLE_POINT_MINUS_P1D}"
   mkdir -p "${SHARE_CYCLE_DIR_MINUS_P1D}"
   echo "I believe in ..." > "${SHARE_CYCLE_DIR_MINUS_P1D}/somefile"

> 

Reviewer: ```suggestion

See ``isodatetime --help`` for usage instructions.

```

paths should be under the *workflow share directory*, the location of which is
passed to all tasks as ``$CYLC_WORKFLOW_SHARE_DIR``.

The ``$CYLC_SHARE_CYCLE_DIR`` variable can be used to target
cyclepoint-specific locations under the workflow share directory.

Sometimes it may be useful to refer to other cycles - to do this use
``isodatetime``:

.. code-block:: bash

   # Cylc task script
   CYCLE_POINT_MINUS_P1D=$(isodatetime "${CYLC_TASK_CYCLE_POINT} --offset -P1D)"
   SHARE_CYCLE_DIR_MINUS_P1D="${CYLC_WORKFLOW_SHARE_DIR}/cycle/${CYCLE_POINT_MINUS_P1D}"
   mkdir -p "${SHARE_CYCLE_DIR_MINUS_P1D}"
   echo "I believe in ..." > "${SHARE_CYCLE_DIR_MINUS_P1D}/somefile"

See ``isodatetime --help`` for usage instructions.

> .. versionadded:: 8.3.1

Reviewer: ```suggestion
.. versionadded:: 8.5.0
```

>    ``$CYLC_WORKFLOW_SHARE_DIR`` and the ``isodatetime`` are designed to

Reviewer: ```suggestion
   ``$CYLC_SHARE_CYCLE_DIR`` (used in combination with the ``isodatetime``
   command are designed to
```

   provide a Cylc internal replacement for the use of ``rose task-env``
   to provide ``$ROSE_DATAC`` and derived variables.

> Subdirectories of ``$CYLC_SHARE_CYCLE_DIR`` etc. should be agreed between

Reviewer: ```suggestion
Subdirectories of ``$CYLC_SHARE_CYCLE_DIR`` should be agreed between
```
